aten/src/ATen/native/cpu/TensorCompareKernel.cpp

```diff
@@ -0,0 +1,106 @@
+#include "ATen/native/cpu/TensorCompareKernel.h"
+
+#include <numeric>
+#include <iterator>
+#include <algorithm>
+
+#include "ATen/Dispatch.h"
+#include "ATen/Parallel.h"
+#include "ATen/optional.h"
+
+namespace at { namespace native { namespace {
+
+template <typename scalar_t>
+bool _isnan(scalar_t val) {
+  return false;
+}
+
+template <>
+bool _isnan(float val) {
+  return std::isnan(val);
+}
+
+template <>
+bool _isnan(double val) {
+  return std::isnan(val);
+}
+
+#define isnan_break(val) \
```

Comment:
I get a bit confused by the control-flow via macro. Can you replace the usages with the if statement?

Reply:
@mingfeima can you make this change? I think this would be good to go afterwards.

Reply:
@SsnL the macro is replaced by if statement now.